AABB_tree/include/CGAL/AABB_segment_primitive_3.h

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2012 INRIA Sophia-Antipolis (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Sebastien Loriot
+//
+
+
+#ifndef CGAL_AABB_SEGMENT_PRIMITIVE_3_H_
+#define CGAL_AABB_SEGMENT_PRIMITIVE_3_H_
+
+#include <CGAL/license/AABB_tree.h>
+
+#include <CGAL/disable_warnings.h>
+
+#include <CGAL/AABB_primitive.h>
+#include <iterator>
+
+namespace CGAL {
+
+namespace internal {
+  template <class GeomTraits, class Iterator>
+  struct Source_of_segment_3_iterator_property_map{
```

Comment:
would be good to factorize these 2D/3D helper classes

Reply:
I had a look with @soesau, and we do not really see what the  common code would be.